### PR TITLE
more precise `BetterNodeFinder` types

### PR DESCRIPTION
**Pull request by staabm**
`PHPParser\NodeFinder` expects a callable returning `bool`, see https://github.com/nikic/PHP-Parser/blob/a6e34665fd6a2bcefc924cb5fd73788767ae510e/lib/PhpParser/NodeFinder.php#L14

### CODE REVIEW

File: src/PhpParser/Node/BetterNodeFinder.php [hunk 1]
```diff
@@ -225,6 +225,7 @@ public function findFirstNonAnonymousClass(array $nodes): ?Node
 
     /**
      * @param Node|Node[] $nodes
+     * @param callable(Node $node): bool $filter
```
Comment (staabm) [inline src/PhpParser/Node/BetterNodeFinder.php:228]: PHPParser\NodeFinder expects a callable returning bools, see https://github.com/nikic/PHP-Parser/blob/a6e34665fd6a2bcefc924cb5fd73788767ae510e/lib/PhpParser/NodeFinder.php#L14

**Comment by staabm** [comment]
ready to go

**Comment by TomasVotruba** [comment]
Thank you :+1:

File: src/PhpParser/Node/BetterNodeFinder.php (hunk 1 above)
Comment (samsonasik) [inline src/PhpParser/Node/BetterNodeFinder.php:228]: @staabm this is somehow make invalid rectify, ref https://github.com/rectorphp/rector-src/pull/1649/commits/620d2b312559b02e0c5fd6bd8f3e1019cc1a2b7d

Comment (staabm) [reply src/PhpParser/Node/BetterNodeFinder.php:228]: Hmm it seems rector is mixing up the parameters

The signature as it was done in this commit is correct

**Pull request merged**